### PR TITLE
Added benchmarks for solving linear matrix systems, Ax=b.

### PR DESCRIPTION
**Pull request by moorepants**

### CODE REVIEW

**Comment by bjodah** [comment]
I think GE, LU and ADJ are good candidates for parametrized tests. See e.g. https://github.com/bjodah/sym/blob/master/benchmarks/benchmarks/Lambdify.py#L61for example of syntax.

**Comment by moorepants** [comment]
I switched it. I think it is correct. Not quite sure.

File: benchmarks/solve.py [hunk 1]
```diff
@@ -54,3 +54,31 @@ def time_solve(self):
 
     def time_solve_nocheck(self):
         sympy.solve(self.eqs, *self.p.c, check=False)
+
+
+class TimeMatrixSolve:
+
+    params = ['GE', 'LU' 'ADJ']
```
Comment (bjodah) [inline benchmarks/solve.py]: there is a comma missing after LU, making the list: 'GE', 'LUADJ'

File: benchmarks/solve.py [hunk 2]
```diff
@@ -54,3 +54,31 @@ def time_solve(self):
...
+        self.b = sympy.Matrix(n, 1,
+                              lambda i, j: sympy.Symbol('b{}{}'.format(i, j)))
+        self.A_sym = sympy.Matrix(n, n, lambda i, j:
+                                  sympy.Symbol('a{}{}'.format(*sorted((i, j)))))
+
+    def time_solve(self, name):
+
+        self.A.solve(self.b, method=name)
+
+    def time_lusolve(self, name):
```
Comment (bjodah) [inline benchmarks/solve.py]: time_lusolve and time_cholesky_solve should be moved to a separate class (since they aren't affected by the parameter)

Comment (moorepants) [inline benchmarks/solve.py]: But they use the same setup. Is the setup run once per method or once per all methods in a class?

Comment (bjodah) [inline benchmarks/solve.py]: once per method, shared code can be put in a private function in the module, e.g. `def _get_A_b_A_sym()`

**Comment by moorepants** [comment]
@bjodah I've updated this. Merge if you think it is fine.

**Comment by bjodah** [comment]
Great!

**Pull request merged**